src/Control/Monad/Catch.hs

```diff
@@ -201,6 +199,24 @@ instance MonadThrow STM where
 instance MonadCatch STM where
   catch = STM.catchSTM
 
+instance e ~ SomeException => MonadThrow (Either e) where
+  throwM = Left . toException
+-- |
```

Comment:
No objection from me, I just typically use this formula for easy extension in the future. I'll update.